src/ray/object_manager/object_buffer_pool.h

```diff
@@ -0,0 +1,171 @@
+  /// connect.
+  /// \param chunk_size The chunk size into which objects are to be split.
+  ObjectBufferPool(const std::string &store_socket_name, const uint64_t chunk_size, const int release_delay);
+
+  ~ObjectBufferPool();
+
+  /// This object cannot be copied due to pool_mutex.
+  RAY_DISALLOW_COPY_AND_ASSIGN(ObjectBufferPool);
+
+  /// \param data_size The size of the object + metadata.
+  /// \return The number of chunks into which the object will be split.
+  uint64_t GetNumChunks(uint64_t data_size);
+
+  /// \param chunk_index The chunk index for which to obtain the buffer length.
+  /// \param data_size The size of the object + metadata.
+  /// \return The number of chunks into which the object will be split.
+  uint64_t GetBufferLength(uint64_t chunk_index, uint64_t data_size);
+
+  /// Returns a chunk of an object at the given chunk_index. The object chunk serves
+  /// as the data that is to be written to a connection as part of sending an object to
+  /// a remote node.
+  ///
+  /// \param object_id The ObjectID.
+  /// \param data_size The sum of the object size and metadata size.
+  /// \param metadata_size The size of the metadata.
+  /// \param chunk_index The index of the chunk.
+  /// \return A pair consisting of a ChunkInfo struct and status of invoking this method.
+  std::pair<const ObjectBufferPool::ChunkInfo &, ray::Status> GetChunk(const ObjectID &object_id, uint64_t data_size,
+                            uint64_t metadata_size, uint64_t chunk_index);
+
+  /// When a chunk is done being used as part of a get, this method is invoked
+  /// to indicate that the buffer associated with a chunk is no longer needed.
+  ///
+  /// \param object_id The object_id of the buffer to release.
+  /// \return The status of invoking this method.
+  ray::Status ReleaseGetChunk(const ObjectID &object_id, uint64_t chunk_index);
+
+  /// Returns a chunk of an empty object at the given chunk_index. The object chunk
+  /// serves as the data that is to be written to by a connection receiving an object
+  /// from a remote node.
+  ///
+  /// \param object_id The ObjectID.
+  /// \param data_size The sum of the object size and metadata size.
+  /// \param metadata_size The size of the metadata.
+  /// \param chunk_index The index of the chunk.
+  /// \return A ChunkInfo struct.
+  std::pair<const ObjectBufferPool::ChunkInfo &, ray::Status> CreateChunk(const ObjectID &object_id, uint64_t data_size,
+                               uint64_t metadata_size, uint64_t chunk_index);
+
+  ray::Status ReleaseCreateChunk(const ObjectID &object_id, uint64_t chunk_index);
+
+  /// Seal the object associated with a create operation.
+  ///
+  /// \param object_id The ObjectID.
+  /// \return The status of invoking this method.
+  ray::Status SealChunk(const ObjectID &object_id, uint64_t chunk_index);
+
+  /// Abort the create operation associated with an object.
+  ray::Status AbortCreate(const ObjectID &object_id);
```

Comment:
This looks like it's only called in the destructor. Should this be private? Also, please be clear in the documentation that this aborts all chunks of the object, including outstanding ones.